ai_replica/utils/system.py

```diff
@@ -0,0 +1,73 @@
+"""Tools for environment- and OS-related stuff"""
+
+import os
+import platform
+import subprocess
+import webbrowser
+
+
+def is_expected_python_version(expected_version):
+    """Returns True if the actual Python version is the same as the expected version, and False otherwise.
+
+    It will also return True if the actual version is a subset of the expected version. For example:
+
+    | expected | actual  | result |
+    | -------- | ------- | ------ |
+    | "3.8.1"  | "3.8.1" | True   |
+    | "3.8"    | "3.8.1" | True   |
+    | "3"      | "3.8.1" | True   |
+    """
+    actual_version = platform.python_version()
+    return actual_version.startswith(expected_version)
+
+
+def create_env_if_not_exist(env_path):
+    """Creates a venv virtual environment"""
+    if not os.path.exists(env_path):
+        subprocess.call(["python3", "-m", "venv", env_path])
```

Comment:
It looks like this script does not work as expected in the following case. The OS has python3.9 installed by default which is accessible via 'python3' command. In addition, python3.8. is installed and is accessible via 'python3.8' command. In this case ''python3 -m venv <path>" command creates python3.9 venv that leads to failure when installing Rasa. But "python3.8 -m venv <path>" works perfectly.

Probably, name of (or path to) python3.8 exec (or 3.7 or whatever version is required by Rasa and installed on the machine) can be provided as an additional parameter to "create_env_if_not_exist" function or set up as env variable which is consumed by create_env_if_not_exist. This env var can be set manually when rasa_launcher.py is launched.

Reply:
Fixed. For the env and the requirements installation, the script now uses the python version defined in config.yaml